File: src/main/java/rocks/sakira/flamingo/register/Items.java
package rocks.sakira.flamingo.register;

import net.minecraft.item.*;
import net.minecraftforge.fml.RegistryObject;
import net.minecraftforge.registries.DeferredRegister;
import net.minecraftforge.registries.ForgeRegistries;
import rocks.sakira.flamingo.Flamingo;
import rocks.sakira.flamingo.item.ItemFlamingoEgg;
import rocks.sakira.flamingo.item.ItemShrimp;
import rocks.sakira.flamingo.item.ItemShrimpCocktail;
import rocks.sakira.flamingo.item.SupplierSpawnEggItem;

public class Items {
    public static final DeferredRegister<Item> REGISTER = DeferredRegister.create(ForgeRegistries.ITEMS, Flamingo.MOD_ID);

    private static final ItemGroup GROUP = new ItemGroup("flamingo_ooo") {
        @Override
        public ItemStack createIcon() {
            return new ItemStack(PINK_FEATHER.get());
        }
    };

    public static final RegistryObject<SpawnEggItem> FLAMINGO_SPAWN_EGG = REGISTER.register(
            "flamingo_spawn_egg",

            () -> new SupplierSpawnEggItem(
                    Entities.FLAMINGO_ENTITY,
                    0xEF9A9A,
                    0xFFE4E4,

                    (new Item.Properties())
                            .group(GROUP)
            )
    );

    public static final RegistryObject<Item> FLAMINGO_EGG = REGISTER.register(
            "flamingo_egg",

            () -> new ItemFlamingoEgg(
                    new Item.Properties()
                            .maxStackSize(16)
                            .group(GROUP)
            )
    );

    public static final RegistryObject<Item> PINK_FEATHER = REGISTER.register(
            "pink_feather",

            () -> new Item(
                    new Item.Properties()
                            .group(GROUP)
            )
    );

    public static final RegistryObject<Item> SHRIMP = REGISTER.register(
            "shrimp",

            () -> new ItemShrimp(
                    new Item.Properties()
                            .maxStackSize(16)
                            .food(Foods.SALMON)  // 2 hunger
                            .group(GROUP)
            )
    );

    public static final RegistryObject<Item> COOKED_SHRIMP = REGISTER.register(
            "cooked_shrimp",

            () -> new Item(
                    new Item.Properties()
                            .maxStackSize(16)
                            .food(Foods.COOKED_RABBIT)  // 5 hunger
                            .group(GROUP)
            )
    );

    public static final RegistryObject<Item> SHRIMP_COCKTAIL = REGISTER.register(
            "shrimp_cocktail",

            () -> new ItemShrimpCocktail(
                    new Item.Properties()
                            .maxStackSize(16)
                            .food(Foods.MUSHROOM_STEW)  // 6 hunger
                            .group(GROUP)
            )
    );

    public static final RegistryObject<Item> SUSPICIOUS_CHICKEN = REGISTER.register(
            "suspicious_chicken",

            () -> new Item(
                    new Item.Properties()
                            .food(Foods.CHICKEN)  // 5 hunger
                            .group(GROUP)
            )
    );

    public static final RegistryObject<Item> COOKED_SUSPICIOUS_CHICKEN = REGISTER.register(
            "cooked_suspicious_chicken",

            () -> new Item(
                    new Item.Properties()
                            .food(Foods.COOKED_CHICKEN)  // 5 hunger
                            .group(GROUP)
            )
    );
}

File: src/main/kotlin/rocks/sakira/flamingo/register/Items.kt
package rocks.sakira.flamingo.register

import net.fabricmc.fabric.api.client.itemgroup.FabricItemGroupBuilder
import net.minecraft.item.*
import net.minecraft.util.Identifier
import net.minecraft.util.registry.Registry
import rocks.sakira.flamingo.MOD_ID
import rocks.sakira.flamingo.item.FlamingoEggItem
import rocks.sakira.flamingo.item.ShrimpCocktailItem
import rocks.sakira.flamingo.item.ShrimpItem

object Items {
    val GROUP: ItemGroup = FabricItemGroupBuilder
        .create(Identifier(MOD_ID, "flamingo"))
        .icon { ItemStack(PINK_FEATHER) }
        .build()

    val PINK_FEATHER = Item(
        Item.Settings().group(GROUP)
    )

    val SHRIMP = ShrimpItem(
        Item.Settings()
            .maxCount(16)
            .food(FoodComponents.SALMON)
            .group(GROUP)
    )

    val COOKED_SHRIMP = Item(
        Item.Settings()
            .maxCount(16)
            .food(FoodComponents.COOKED_RABBIT)
            .group(GROUP)
    )

    val SHRIMP_COCKTAIL = ShrimpCocktailItem(
        Item.Settings()
            .maxCount(16)
            .food(FoodComponents.MUSHROOM_STEW)
            .group(GROUP)
    )

    val SUSPICIOUS_CHICKEN = Item(
        Item.Settings()
            .food(FoodComponents.CHICKEN)
            .group(GROUP)
    )

    val COOKED_SUSPICIOUS_CHICKEN = Item(
        Item.Settings()
            .food(FoodComponents.COOKED_CHICKEN)
            .group(GROUP)
    )

    val FLAMINGO_SPAWN_EGG = SpawnEggItem(
        Entities.FLAMINGO,
        0xEF9A9A,
        0xFFE4E4,
        Item.Settings().group(GROUP)
    )

    val FLAMINGO_EGG = FlamingoEggItem(
        Item.Settings().group(GROUP)
    )

    fun register() {
        Registry.register(Registry.ITEM, Identifier(MOD_ID, "pink_feather"), PINK_FEATHER)

        Registry.register(Registry.ITEM, Identifier(MOD_ID, "shrimp"), SHRIMP)
        Registry.register(Registry.ITEM, Identifier(MOD_ID, "cooked_shrimp"), COOKED_SHRIMP)
        Registry.register(Registry.ITEM, Identifier(MOD_ID, "shrimp_cocktail"), SHRIMP_COCKTAIL)

        Registry.register(Registry.ITEM, Identifier(MOD_ID, "suspicious_chicken"), SUSPICIOUS_CHICKEN)
        Registry.register(Registry.ITEM, Identifier(MOD_ID, "cooked_suspicious_chicken"), COOKED_SUSPICIOUS_CHICKEN)

        Registry.register(Registry.ITEM, Identifier(MOD_ID, "flamingo_spawn_egg"), FLAMINGO_SPAWN_EGG)
        Registry.register(Registry.ITEM, Identifier(MOD_ID, "flamingo_egg"), FLAMINGO_EGG)
    }
}
